File: algorithms/Selection_Sort.py
# We need the time module to create some time difference between each comparison
import time

# Importing colors from colors.py
from Helper.color import *

def selection_sort(data, drawData, timeTick):
    for i in range(len(data) - 1):
        Min_Idx = i
        for k in range(i + 1, len(data)):
            if data[k] < data[Min_Idx]:
                Min_Idx = k

        data[Min_Idx], data[i] = data[i], data[Min_Idx]
        drawData(data, [YELLOW if x == Min_Idx or x == i else BLUE for x in range(len(data))])
        time.sleep(timeTick)
        
    drawData(data, [BLUE for x in range(len(data))])

File: Algorithms_Code/JavaScript/Selection_Sort.js
const Selection_Sort = (arr) => {
    const n = arr.length;

    for (let i = 0; i < n - 1; i++) {
        let min_idx = i;

        for (let j = i + 1; j < n; j++) {
            if (arr[j] < arr[min_idx]) min_idx = j;
        }

        [arr[min_idx], arr[i]] = [arr[i], arr[min_idx]];
    }
}